### PR TITLE
feat(date picker day): add date picker day component => DEV

### PR DESCRIPTION
**Pull request by Sarath-Kumar-G**
# PR Template & Definition of Done

## What does this PR do?

- Add `DatePickerDay` component
- Added custom-style classes for days in the calendar  

## What steps does your reviewer have to take to test this PR manually?

1. Check the Calander Days by changing the `DaysContent` params
2. Check the customization by passing different props

## Testing checklist - Please check off

- [ ] I have written unit tests for all components files I've worked on, these tests cover what the user sees and how the UI changes when they interact with it.
- [ ] I have written unit tests for all helper functions I've worked on, that cover use-cases and edge-cases.
- [ ] I have written integration tests for features I've worked on, to ensure my components work together as I expect.

### If you have not added tests for any of the above, please explain why below.

Jest config is not merged into dev.

## Definition of Done - Please check off

- [ ] I have completed the above testing checklist so my code is well tested and I have confidence my code works as I expect in a variety of situations.
- [X] I have Eslint and Prettier enabled to run on file save and I have fixed all errors highlighted by Eslint.
- [X] I have added complete Prop Types and type definitions without using `any` for all my components and helper functions.
- [X] I have deleted all non-descriptive comments and dead code from the files I've touched.
- [X] I have rebased my branch with the base branch I want to merge into and all the commits in this PR are my own.

### CODE REVIEW

File: src/components/atoms/DatePicker/DatePickerDay/DatePickerDay.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  dayLabel: string;
+  isSelectedDate: boolean;
+}
+export function DatePickerDay({
+  day,
+  date,
+  dayLabel,
+  isSelectedDate,
+}: DatePickerDayProps) {
+  console.log({ day, date, dayLabel, isSelectedDate });
```
Comment (sachinatcodewalnut) [inline src/components/atoms/DatePicker/DatePickerDay/DatePickerDay.tsx:16]: remove the logs;

Comment (Sarath-Kumar-G) [reply src/components/atoms/DatePicker/DatePickerDay/DatePickerDay.tsx]: thank removed.

**Review by sachinatcodewalnut — APPROVED** [review]
